connexion/apis/abstract.py

```diff
@@ -283,6 +314,20 @@ def load_spec_from_file(self, arguments, specification):
             swagger_string = jinja2.Template(swagger_template).render(**arguments)
             return yaml.safe_load(swagger_string)  # type: dict
 
+    def _get_spec_version(self, spec):
+        try:
+            version_string = spec.get('openapi') or spec.get('swagger')
```

Comment:
try:
  spec['openapi'] or spec['swagger']
except (KeyError, AttributeError):
  raise ...

Reply:
That has slightly different behavior. This block in my code is checking that (a) spec is a dictionary, and (b) getting the version string from either version of the spec.
Another way to do it would be to check against each version serially, but it's a bit more nested:
```
try:
    version_string = spec['openapi']
except (KeyError, AttributeError):
    try:
        version_string = spec["swagger"]
    except (KeyError, AttributeError):
        raise ...
```

Reply:
OT: @dtkav  thank you very much for all your work in supporting OpenAPI v3 :)